model/jpa/src/main/java/org/keycloak/organization/jpa/JpaOrganizationProvider.java

```diff
@@ -302,12 +303,11 @@ private TypedQuery<Long> buildCountQuery(CriteriaBuilder builder, CriteriaQuery<
 
     private Predicate buildStringSearchPredicate(CriteriaBuilder builder, CriteriaQuery<?> query, Root<OrganizationEntity> org, String search,
                                                  Boolean exact) {
-        Root<OrganizationDomainEntity> domain = query.from(OrganizationDomainEntity.class);
+        Join<OrganizationEntity, OrganizationDomainEntity> domain = org.join("domains", JoinType.LEFT);
```

Comment:
We don't need this join, do we?

Reply:
The LEFT join ensures organizations without domains are included while the domain predicate is applied.

Reply:
My point is that queries will only operate on the organization table, as we don't have, AFAIK, any filter or anything else that will end up in an inner join on the domain table. Therefore, we don't need a left join at all?